clang/lib/Sema/CheckExprLifetime.h

```diff
@@ -18,29 +18,42 @@
 
 namespace clang::sema {
 
-/// Describes an entity that is being assigned.
-struct AssignedEntity {
-  // The left-hand side expression of the assignment.
-  Expr *LHS = nullptr;
+struct CapturingEntity {
```

Comment:
I think that using CapturingEntity for both use cases, with one of them keeping a field null makes it much harder to follow.
Is it possible to use a more specific type?

Reply:
+1. 

I'd not change the existing the `AssignedEntity`, as this structure is designed for the assignment case.

It looks like for the capture_by case, we only need the `Expression`, I think we could use `Expr*` directly in the function parameter, e.g. `void checkCaptureLifetime(Sema &SemaRef, const Expr* Entity, Expr *Captured);`.